coin/coin.go

```diff
@@ -128,10 +132,20 @@ func mul64(a, b int64) (int64, error) {
 // currencies, or if the combination would cause
 // an overflow
 func (c Coin) Add(o Coin) (Coin, error) {
+	// If any of the coins represents no value and does not have a ticker
```

Comment:
Q:  a coin without a ticker is an invalid coin. even if it is zero, IMHO. Is this code due to https://github.com/iov-one/weave/issues/369 not done yet?

Reply:
That is a good point. 

This code is because of https://github.com/iov-one/weave/pull/376/files#diff-e1c2020236c87568b4fce1d6a33e620cR39

It is often in weave codebase that instead of a pointer a zero value is returned and later used, for example `weave.CheckResult` or mentioned `coin.Coin`. If they are returned together with an `error` value, they are not a pointer but an empty structure.
The above cause sometimes weird situations where a value is zero (not `nil`). Because it is not `nil`, all methods should apply (it is a value right?). And then you are getting into a complicated question of "does this make sense?" :grin: 

In this case I think it does make sense. My logic was
```
0          + 1 BTC = 1 BTC  // zero value
0 BTC + 1 BTC = 1 BTC
0 BTC + 0 ETH = error
1 BTC + 0 ETH = error
0 BTC + 1 ETH = error
```

Reply:
I agree with the zero value, uninitialized struct.
Since we don't use pointers everywhere, and we mean 'noop' / 'do nothing' when returning coin.Coin{}, we can make the functions treat it special